src/build_backend.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+import hashlib
+import os
+import shutil
+import subprocess
+import tarfile
+import zipfile
+
+from typing import Dict, Iterable, Optional
+
+from cockpit import __version__
+
+PACKAGE = f'cockpit-{__version__}'
+TAG = 'py3-none-any'
+
+
+def find_sources(srcpkg: bool) -> Iterable[str]:
+    try:
+        subprocess.check_call(['modules/checkout'], stdout=2)
+    except FileNotFoundError:
```

Comment:
When would that script not exist? For out-of-tree builds? If that fails, it would silently continue and probably cause a failure later on -- likely even at runtime, where ferny and systemd_ctypes didn't get installed?

Reply:
It doesn't exist in two cases:
 - when building from a cockpit tarball release
 - when building from an unpacked sdist (which happens when you do `build .`, for example)

Reply:
Could this become more explicit, and check if src/_vendor/ferny exists? either with an `if` (and no `except`), or as an assertion in the `except:` case (probably better to ensure it always checks out the latest version)

Reply:
Doing the check first would be unpythonic.

I'm happy to add the assertion in the `except:` clause.

Probably a comment makes sense, too.

Reply:
Added both.  I put the assert outside of the `except:` block because the file should exist in both cases.

Reply:
Thanks, really nice! Sorry for nagging..